Frontend/CarRentalAdmin/src/app/_services/auth.guard.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import {
+    ActivatedRouteSnapshot,
+    Router,
+    RouterStateSnapshot,
+    CanActivate
+} from '@angular/router';
+
+import { AuthService } from './auth.service';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+    constructor(
+        private authService: AuthService,
+        private router: Router
+    ) {
+    }
+
+    public canActivate(
+        route: ActivatedRouteSnapshot,
+        state: RouterStateSnapshot
+    ): boolean {
+        if (this.authService.isLoggedIn()) {
+            return true;
+        }
+
+        this.router.navigate(
+            ['login'],
+            {queryParams: {returnUrl: state.url}}
```

Comment:
```suggestion
            { queryParams: { returnUrl: state.url } }
```

for better readability